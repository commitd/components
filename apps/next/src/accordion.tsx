import {
  Accordion,
  AccordionContent,
  AccordionHeader,
  AccordionItem,
} from '@committed/ds'

export const AccordionExample = () => (
  <Accordion type="multiple">
    <AccordionItem value="item-1">
      <AccordionHeader>Item 1</AccordionHeader>
      <AccordionContent>
        Here goes the content for the accordion item 1.
      </AccordionContent>
    </AccordionItem>
    <AccordionItem value="item-2">
      <AccordionHeader>Item 2</AccordionHeader>
      <AccordionContent>
        Here goes the content for the accordion item 2.
      </AccordionContent>
    </AccordionItem>
    <AccordionItem value="item-3">
      <AccordionHeader>Item 3</AccordionHeader>
      <AccordionContent>
        Here goes the content for the accordion item 3.
      </AccordionContent>
    </AccordionItem>
  </Accordion>
)
