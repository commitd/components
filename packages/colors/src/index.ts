export { presetColors } from './preset'
export type { Color } from './preset'
