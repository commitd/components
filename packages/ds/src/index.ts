// component exports
export * from './components'
export * from './utils'

// reexports from ss
export { css, styled } from '@committed/ds-ss'
