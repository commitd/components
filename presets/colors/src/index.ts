export {
  neutralColors,
  preset as presetColors,
  primaryColors,
  semanticColors,
} from './preset'
export type { Color, Prefix, SemanticColor } from './preset'
