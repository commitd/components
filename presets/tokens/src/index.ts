export { preset as presetTokens } from './preset'
