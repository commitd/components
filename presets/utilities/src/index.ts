export { preset as presetUtilities, surfaces } from './preset'
export type { Surface, SurfaceVariants } from './preset'

