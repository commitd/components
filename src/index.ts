export * from './components'
export * from './theme'
export * from './hooks'

// Reexport
export { styled } from '@material-ui/styles'

// export shortcodes for convenience
import * as shortcodes from './components'
export { shortcodes }
