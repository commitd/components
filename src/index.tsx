export * from './components'
export * from './theme'
export * from './styles'
export * from './util'
// Some explicit exports are required to resolve ambiguity
export { ThemeProvider, colors } from './theme'
export { makeStyles, styled, useTheme, withTheme } from './styles'
export type { ThemeProviderProps } from './theme'
export type { ComponentCreator, WithTheme } from './styles'
