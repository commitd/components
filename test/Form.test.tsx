import React from 'react'
import { renderLight, renderDark, fireEvent } from './setupTests'
import { Form, TextField, Button } from '../src'

const Example = ({ onSubmit }: { onSubmit: (e: any) => void }) => (
  <Form width={1 / 2} onSubmit={onSubmit} data-testid="form">
    <TextField id="standard-name" label="Name" margin="normal" />
    <Button type="submit" color="primary" mt={2}>
      Submit
    </Button>
  </Form>
)

it('renders light without crashing', () => {
  const { asFragment } = renderLight(<Form />)
  expect(asFragment()).toMatchSnapshot()
})

it('renders dark without crashing', () => {
  const { asFragment } = renderDark(<Form />)
  expect(asFragment()).toMatchSnapshot()
})

it('renders dark without crashing', () => {
  const onSubmit = jest.fn()
  const { getByTestId } = renderDark(<Example onSubmit={onSubmit} />)
  fireEvent.submit(getByTestId('form'))
  expect(onSubmit).toHaveBeenCalled()
})
