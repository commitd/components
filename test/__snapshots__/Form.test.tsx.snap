// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dark without crashing 1`] = `
<DocumentFragment>
  <form
    class="ForwardRef(withPositioning)-root ForwardRef(withPositioning)-root ForwardRef(withSizing)-root ForwardRef(withSizing)-root"
  />
</DocumentFragment>
`;

exports[`renders light without crashing 1`] = `
<DocumentFragment>
  <form
    class="ForwardRef(withPositioning)-root ForwardRef(withPositioning)-root ForwardRef(withSizing)-root ForwardRef(withSizing)-root"
  />
</DocumentFragment>
`;
